While I try to mirror the same test from L4 to A100, it seems huggingface generation of `meta-llama/Llama-2-7b-hf` is incorrect, see https://buildkite.com/vllm/ci-aws/builds/1994#0190318c-8cda-4ce3-9445-36a5ab48d3dd for example. The hf output is just repeating the prompt.

Locally, I can reproduce this issue with A100, but not H100. I extended the output tokens to 50:
```
>           assert hf_output_str == vllm_output_str, (
                f"Test{i}:\nHF: {hf_output_str!r}\nvLLM: {vllm_output_str!r}")
E           AssertionError: Test5:
E             HF: 'Analyze the impact of the COVID-19 pandemic on global economic structures and future business models.\nAnalyze the impact of the COVID-19 pandemic on global economic structures and future business models. In a 1,000-1,250-word paper, address the following:\nAnalyze the impact'
E             vLLM: 'Analyze the impact of the COVID-19 pandemic on global economic structures and future business models.\nThe COVID-19 pandemic has had a significant impact on global economic structures and future business models. The pandemic has caused significant disruptions to supply chains, labor markets, and financial systems, leading to increased uncertainty and risk for'
E           assert 'Analyze the ...ze the impact' == 'Analyze the ... and risk for'
E             
E               Analyze the impact of the COVID-19 pandemic on global economic structures and future business models.
E             - The COVID-19 pandemic has had a significant impact on global economic structures and future business models. The pandemic has caused significant disruptions to supply chains, labor markets, and financial systems, leading to increased uncertainty and risk for
E             + Analyze the impact of the COVID-19 pandemic on global economic structures and future business models. In a 1,000-1,250-word paper, address the following:
E             + Analyze the impact

distributed/test_basic_distributed_correctness.py:59: AssertionError
```

And I find vLLM output is better.

I suspect there are some bugs in hf implementation for llama2-7b, which is fixed in vLLM.

--- a/.buildkite/test-pipeline.yaml
+++ b/.buildkite/test-pipeline.yaml
@@ -185,6 +185,8 @@ steps:
 - label: Distributed Tests (A100)
   gpu: a100
   commands: 
+  # NOTE: don't test llama model here, it seems hf implementation is buggy
+  # see https://github.com/vllm-project/vllm/pull/5689 for details
   - pytest -v -s distributed/test_custom_all_reduce.py
   - TEST_DIST_MODEL=facebook/opt-125m DISTRIBUTED_EXECUTOR_BACKEND=ray pytest -v -s distributed/test_basic_distributed_correctness.py
   - TEST_DIST_MODEL=facebook/opt-125m DISTRIBUTED_EXECUTOR_BACKEND=mp pytest -v -s distributed/test_basic_distributed_correctness.py

